src/network.js

```diff
@@ -170,47 +173,35 @@ class Network {
    * @private
    */
   _writeMessage (conn, msg) {
-    return new Promise((resolve, reject) => {
-      pull(
-        pull.values([msg]),
-        lp.encode(),
-        conn,
-        pull.onEnd((err) => {
-          if (err) return reject(err)
-          resolve()
-        })
-      )
-    })
+    return pipe(
+      [msg],
+      lp.encode(),
+      conn
+    )
   }
 }
 
-function writeReadMessage (conn, msg) {
-  return new Promise((resolve, reject) => {
-    pull(
-      pull.values([msg]),
-      lp.encode(),
-      conn,
-      pull.filter((msg) => msg.length < c.maxMessageSize),
-      lp.decode(),
-      pull.collect((err, res) => {
-        if (err) {
-          return reject(err)
-        }
-        if (res.length === 0) {
-          return reject(errcode(new Error('No message received'), 'ERR_NO_MESSAGE_RECEIVED'))
-        }
-
-        let response
-        try {
-          response = Message.deserialize(res[0])
-        } catch (err) {
-          return reject(errcode(err, 'ERR_FAILED_DESERIALIZE_RESPONSE'))
-        }
-
-        resolve(response)
-      })
-    )
-  })
+async function writeReadMessage (conn, msg) {
+  const res = await pipe(
+    [msg],
+    lp.encode(),
+    conn,
+    utils.itFilter(
+      (msg) => msg.length < c.maxMessageSize
+    ),
+    lp.decode(),
+    async source => {
+      for await (const chunk of source) {
```

Comment:
Not sure if it's worth adding a dependency just for this, but the streaming iterables library has some useful functions, eg this could be replaced with a [map](https://github.com/bustle/streaming-iterables#map) and a collect